src/test/bctest.py

```diff
@@ -6,6 +6,17 @@
 import os
 import json
 import sys
+import binascii
+
+def compare_output(a, b, fmt):
+	'''Data-format dependent output comparison.'''
+	if fmt == 'json': # json: compare parsed data
+		match = (json.loads(a) == json.loads(b))
+	elif fmt == 'hex': # hex: parse and compare binary data
+		match = (binascii.a2b_hex(a.rstrip()) == binascii.a2b_hex(b.rstrip()))
```

Comment:
Why is this rstripped rather than stripped? Is there a case where you want to fail here if leading whitespace differs?